x-pack/plugins/ml/server/lib/register_settings.ts

```diff
@@ -30,5 +34,43 @@ export function registerKibanaSettings(coreSetup: CoreSetup) {
         }),
       },
     },
+    [ANOMALY_DETECTION_ENABLE_TIME_RANGE]: {
+      name: i18n.translate('xpack.ml.advancedSettings.enableAnomalyDetectionDefaultTimeRangeName', {
+        defaultMessage: 'Enable time filter defaults for anomaly detection jobs',
+      }),
+      value: false,
+      schema: schema.boolean(),
+      description: i18n.translate(
+        'xpack.ml.advancedSettings.enableAnomalyDetectionDefaultTimeRangeDesc',
+        {
+          defaultMessage:
+            'Use the default time filter in the Single Metric Viewer and Anomaly Explorer.',
```

Comment:
I think this needs an extra sentence to explain what happens if this is not enabled. Something like `If not enabled, the results for the full time range of the job will be displayed.`

Reply:
> Something like ... "If not enabled, the results....

LGTM, except I'd replace "will be" with "are".

Reply:
Updated it to `Use the default time filter in the Single Metric Viewer and Anomaly Explorer. If not enabled, the results for the full time range of the job are displayed.` here